### PR TITLE
feat: Add Graph Connector sample app

### PR DESCRIPTION
**Pull request by formulahendry**
Previous PR: https://github.com/OfficeDev/TeamsFx-Samples/pull/335

Demo:
graph-connector-app-demo

Architecture:
image

### CODE REVIEW

File: graph-connector-app/api/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+## Prerequisites
+
+- [NodeJS](https://nodejs.org/en/)
+- An M365 account. If you do not have M365 account, apply one from [M365 developer program](https://developer.microsoft.com/en-us/microsoft-365/dev-program)
+- [Teams Toolkit Visual Studio Code Extension](https://aka.ms/teams-toolkit) or [TeamsFx CLI](https://aka.ms/teamsfx-cli)
+
+## Develop
+
+The Teams Toolkit IDE Extension and TeamsFx CLI provide template code for you to get started with Azure Functions for your Teams application. Microsoft Teams Framework simplifies the task of establishing the user's identity within the Azure Function.
```
Comment (SLdragon) [inline graph-connector-app/api/README.md:13]: Teams Toolkit IDE Extension -> Teams Toolkit Extension？

File: graph-connector-app/api/data/index.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      skip_empty_lines: true
+    });
+    const graphClient: Client = createMicrosoftGraphClient(teamsfx);
+    for (const item of records) {
+      await graphClient.api(`/external/connections/${connectionId}/items/${item.PartNumber}`)
+        .put({
+          "acl": [
+            {
+              "type": "everyone",
+              "value": "c5f19b2d-0a77-454a-9b43-abf298c3b34e",
```
Comment (blackchoey) [inline graph-connector-app/api/data/index.ts:67]: should we generate a random value here?

Comment (formulahendry) [reply graph-connector-app/api/data/index.ts:67]: In this sample, we did not have special permission control. However, the `acl` is a required property in this API. So just leave a random value.

File: graph-connector-app/api/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## Debug
+
+- From Visual Studio Code: Start debugging the project by hitting the `F5` key in Visual Studio Code. Alternatively use the `Run and Debug Activity Panel` in Visual Studio Code and click the `Start Debugging` green arrow button.
+- From TeamsFx CLI: Start debugging the project by executing the command `teamsfx preview --local` in your project directory.
+
+## Edit the manifest
+
+You can find the Teams app manifest in `templates/appPackage` folder. The folder contains two manifest files:
+* `manifest.local.template.json`: Manifest file for Teams app running locally.
```
Comment (blackchoey) [inline graph-connector-app/api/README.md:61]: do we need to keep this readme.md? Something like this needs to be updated since our toolkit will change its behavior in next build. If we do not want to keep maintaining this file, we may remove it.

File: graph-connector-app/api/proxies.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (adashen) [inline graph-connector-app/api/proxies.json:1]: is this file necessary?

File: graph-connector-app/api/README.md (hunk 3 above)
Comment (formulahendry) [reply graph-connector-app/api/README.md]: Thanks for catching this! Remove README file of `tabs` and `api` to align with other samples.

**Review by blackchoey — APPROVED** [review]

File: graph-connector-app/api/proxies.json (hunk 4 above)
Comment (formulahendry) [reply graph-connector-app/api/proxies.json:1]: It is generated from default template, and other sample keep it as well. So just keep it with no change.

**Review by SLdragon — APPROVED** [review]
